flocker/node/_deploy.py

```diff
@@ -730,18 +766,25 @@ def calculate_necessary_state_changes(self, local_state,
             desired_local_state)
 
         start_containers = [
-            StartApplication(application=app, hostname=self.hostname)
+            StartApplication(application=app, node_state=current_node_state)
             for app in desired_node_applications
-            if app.name in start_names
+            if ((app.name in start_names) and
+                # If manifestation isn't available yet, don't start:
+                # XXX in FLOC-1240 non-primaries should be checked.
+                (app.volume is None or
+                 app.volume.manifestation.dataset_id in
+                 current_node_state.manifestations))
```

Comment:
:question: This would probably make a good stand-alone function - `app_volumes_available(app, current_node_state)` or some such.